### DISCUSSION TITLE
Move addons/panku_console/ to a separate repo to use as git submodules

### DISCUSSION THREAD

**Discussion by 0xIonRod** [Ark2000/PankuConsole#104] [Ideas]
By doing that it'll be much easier to reuse to code in others projects as it'll require only to add a new repo as a submodule

**Comment by Ark2000** [comment]
It sounds good, is there any godot plugin project examples on Github I can follow?

**Reply by 0xIonRod** [reply]
Yep, for example this one [https://github.com/4d49/object-inspector](https://github.com/4d49/object-inspector)

**Reply by Ark2000** [reply]
I see, I will create another mirror repo that only contains addons folder later. The Godot built-in plugins manager is pretty limited, https://github.com/godotengine/godot-proposals/issues/2262, maybe there’s a chance we can develop an external plugin manager?

**Reply by 0xIonRod** [reply]
I've tried to make one a while ago: [https://github.com/LunCoSim/godot-package-manager](https://github.com/LunCoSim/godot-package-manager)

Aaaandd... It sucked. 

For now the solution with submodules looks decent, I've described it here: [How to structure Godot Plugins folder for convinient development?](https://github.com/LunCoSim/lunco-sim/blob/main/docs/LunCo%20Docs/LunCo-Architecture.md#how-to-structure-godot-plugins-folder-for-convinient-development)


For now the main issue with submodules - it has to be added with ssh link to be able to push, meaning you'll need github account.

http link doesnt require github account, but you can't push.


I'm thinking of making more general godot plugins management system, that would allow to easilly distribute UGC on the fly

Happy to discuss in details on [Twitter](https://twitter.com/_Difint_)

**Reply by scriptsengineer** [reply]
I loved the discussion, as a user that came from unity, I used package manager along with submodules, I wanted to know @Ark2000 how you made your pipeline to create a mirror repo?

@Difint  the solution would be viable if at least they removed the need to put addons inside and this submodule download process move direct to addons, if a module uses a folder outside the addons it must be an exception.

**Reply by Ark2000** [reply]
Currently I update the mirror repo manually, but it's expected to automate this process eventually using something like github actions, local script etc.

**Comment by univeous** [comment]
I'm using [gd-plug](https://github.com/imjp94/gd-plug) and I think it works great!
